app/src/common/shared/com/igalia/wolvic/VRBrowserActivity.java

```diff
@@ -411,16 +406,19 @@ private void attachToWindow(@NonNull WindowWidget aWindow, @Nullable WindowWidge
         return EngineProvider.INSTANCE.getOrCreateRuntime(this).getCrashReportIntent();
     }
 
-    private void showPrivacyDialogIfNeeded() {
+    // Returns true if the dialog was shown, false otherwise.
+    private boolean showPrivacyDialogIfNeeded() {
         if (SettingsStore.getInstance(this).isPrivacyPolicyAccepted()) {
-            return;
+            return false;
         }
 
         PrivacyPolicyDialogWidget privacyPolicyDialog = new PrivacyPolicyDialogWidget(this);
         privacyPolicyDialog.setDelegate(response -> {
             if (response) {
                 Log.w(LOGTAG, "The user accepted the privacy policy.");
                 SettingsStore.getInstance(this).setPrivacyPolicyAccepted(true);
+                showWhatsNewDialogIfNeeded();
+                mWindows.restoreSessions();
```

Comment:
It's unfortunate to have this same code twice, not because we're repeating code, but because we have to maintain these two code paths and implement future new behaviours twice. Not sure how to improve that, perhaps we could start by just refactoring those two lines in an utility method...

Reply:
I have put the call to `mWindows.restoreSessions()` back at its original position at the end of `initializeWidgets()`, so now it is independent of the first launch dialogs.